src/CorePolicy.cc

```diff
@@ -0,0 +1,247 @@
+/* Copyright (c) 2017 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <thread>
+#include "PerfUtils/TimeTrace.h"
+#include "PerfUtils/Util.h"
+#include "PerfUtils/Cycles.h"
+#include "CoreArbiter/CoreArbiterClient.h"
+
+#include "Arachne.h"
+#include "CorePolicy.h"
+#include "PerfStats.h"
+#include "CoreArbiter/ArbiterClientShim.h"
+
+using PerfUtils::Cycles;
+
+/*
+ * We will attempt to increase the number of cores if the idle core
+ * fraction (computed as idlePercentage * numSharedCores) is less than this
+ * number.
+ */
+const double maxIdleCoreFraction = 0.1;
+
+/*
+ * We will attempt to increase the number of cores if the load factor
+ * increases beyond this threshold.
+ */
+const double loadFactorThreshold = 1.0;
+
+/*
+ * Save the core fraction at which we ramped up based on load factor, so we
+ * can decide whether to ramp down.  Allocated in bootstrapLoadEstimator.

```

Comment:
Not enough information. This is an array, right? What are the indices?

Also, I'm not sure that "core fraction" is quite specific enough; is that the same as "idle core fraction"? If so, I would use the same term.